tests/manager/repositories/container_registry/test_container_registry_repository.py

```diff
@@ -1,233 +1,921 @@
-        self, repository, mock_db_readonly_session, sample_registry_row
-    ):
+        self, repository: ContainerRegistryRepository, database_engine: ExtendedAsyncSAEngine
+    ) -> None:
         """Test successful registry retrieval by name and project"""
-        mock_db_readonly_session.scalar.return_value = sample_registry_row
-
-        result = await repository.get_by_registry_and_project(
-            "registry.example.com", "test-project"
-        )
-
-        assert result is not None
-        assert isinstance(result, ContainerRegistryData)
-        assert result.registry_name == "registry.example.com"
-        assert result.project == "test-project"
-
-        # Verify the query was built correctly
-        mock_db_readonly_session.scalar.assert_called_once()
+        project_name = "test-project"
+        registry_name = "test-registry.example.com"
+        async with self.create_test_registry(
+            database_engine, registry_name, project_name
+        ) as registry_data:
+            # When
+            result = await repository.get_by_registry_and_project(registry_name, project_name)
+
+            # Then
+            assert result is not None
+            assert isinstance(result, ContainerRegistryData)
+            assert result.registry_name == registry_name
+            assert result.project == project_name
+            assert result.id == registry_data.id
 
     @pytest.mark.asyncio
     async def test_get_by_registry_and_project_not_found(
-        self, repository, mock_db_readonly_session
-    ):
-        """Test registry retrieval when not found"""
-        mock_db_readonly_session.scalar.return_value = None
-
+        self, repository: ContainerRegistryRepository
+    ) -> None:
+        """Test registry retrieval when registry doesn't exist"""
         with pytest.raises(ContainerRegistryNotFound):
             await repository.get_by_registry_and_project("non-existent", "project")
 
     @pytest.mark.asyncio
     async def test_get_by_registry_name(
-        self, repository, mock_db_readonly_session, sample_registry_row
-    ):
-        """Test getting all registries with a specific name"""
-        mock_result = MagicMock()
-        mock_result.scalars.return_value.all.return_value = [sample_registry_row]
-        mock_db_readonly_session.execute.return_value = mock_result
-
-        result = await repository.get_by_registry_name("registry.example.com")
+        self, repository: ContainerRegistryRepository, database_engine: ExtendedAsyncSAEngine
+    ) -> None:
+        """Test retrieving all registries with the same name"""
+        registry_name = "test-registry.example.com"
+        project1_name = "project1"
+        project2_name = "project2"
+
+        async with (
+            self.create_test_registry(database_engine, registry_name, project1_name),
+            self.create_test_registry(database_engine, registry_name, project2_name),
+        ):
```

Comment:
This should have already been injected through a fixture, and names like project1, 2 should not be arbitrary values but should be generated through random values and injected by the fixture.
If you write test code like this, it may break due to timing issues with other test code, and the logic that the test code is testing is obscured and not easily visible.